packages/cli/src/utils.ts

```diff
@@ -25,3 +25,20 @@ export function getIdentifierName(value: string) {
 
   return sanitisedValue;
 }
+
+/**
+ * Get the defined values from an array, removing all `undefined` values. If the
+ * array itself is `undefined`, an empty array is returned.
+ *
+ * @param array - The array to get the defined values from.
+ * @returns The array with all `undefined` values removed.
+ */
+export function getDefinedArray<Type>(
+  array: readonly (Type | undefined)[] | undefined,
+): Type[] {
+  if (!array) {
+    return [];
+  }
+
+  return array.filter((value): value is Type => value !== undefined);
```

Comment:
This seems a bit redundant and potentially misleading, in that we are not actually any more certain that the values of the original array are of type `(Type | undefined)[]` after this function has been called.
```suggestion
  return array.filter((value) => value !== undefined);
```

Reply:
It's not redundant since `filter` does not infer the type without it.

```ts
declare const array: (string | undefined)[];
array.filter((value) => value !== undefined); // (string | undefined)[]
array.filter((value): value is string => value !== undefined); // string[]
```

Removing it defeats the purpose of the function. 😅

Reply:
lol, I overestimated `.filter()`. Never mind!